spec.html

```diff
@@ -47724,26 +47724,26 @@ <h1>unescape ( _string_ )</h1>
         <p>It performs the following steps when called:</p>
         <emu-alg>
           1. Set _string_ to ? ToString(_string_).
-          1. Let _length_ be the length of _string_.
+          1. Let _len_ be the length of _string_.
           1. Let _R_ be the empty String.
           1. Let _k_ be 0.
-          1. Repeat, while _k_ ≠ _length_,
-            1. Let _c_ be the code unit at index _k_ within _string_.
-            1. If _c_ is the code unit 0x0025 (PERCENT SIGN), then
-              1. Let _hexEscape_ be the empty String.
-              1. Let _skip_ be 0.
-              1. If _k_ ≤ _length_ - 6 and the code unit at index _k_ + 1 within _string_ is the code unit 0x0075 (LATIN SMALL LETTER U), then
-                1. Set _hexEscape_ to the substring of _string_ from _k_ + 2 to _k_ + 6.
-                1. Set _skip_ to 5.
-              1. Else if _k_ ≤ _length_ - 3, then
-                1. Set _hexEscape_ to the substring of _string_ from _k_ + 1 to _k_ + 3.
-                1. Set _skip_ to 2.
-              1. If _hexEscape_ can be interpreted as an expansion of |HexDigits[~Sep]|, then
-                1. Let _hexIntegerLiteral_ be the string-concatenation of *"0x"* and _hexEscape_.
-                1. Let _n_ be ! ToNumber(_hexIntegerLiteral_).
-                1. Set _c_ to the code unit whose value is ℝ(_n_).
-                1. Set _k_ to _k_ + _skip_.
-            1. Set _R_ to the string-concatenation of _R_ and _c_.
+          1. Repeat, while _k_ &lt; _len_,
+            1. Let _C_ be the code unit at index _k_ within _string_.
+            1. If _C_ is the code unit 0x0025 (PERCENT SIGN), then
+              1. Let _hexDigits_ be the empty String.
+              1. Let _optionalAdvance_ be 0.
+              1. If _k_ + 5 &lt; _len_ and the code unit at index _k_ + 1 within _string_ is the code unit 0x0075 (LATIN SMALL LETTER U), then
+                1. Set _hexDigits_ to the substring of _string_ from _k_ + 2 to _k_ + 6.
+                1. Set _optionalAdvance_ to 5.
+              1. Else if _k_ + 2 &lt; _len_, then
+                1. Set _hexDigits_ to the substring of _string_ from _k_ + 1 to _k_ + 3.
+                1. Set _optionalAdvance_ to 2.
+              1. Let _parseResult_ be ParseText(StringToCodePoints(_hexDigits_), |HexDigits[~Sep]|).
+              1. If _parseResult_ is a Parse Node, then
+                1. Let _n_ be the MV of _parseResult_.
+                1. Set _C_ to the code unit whose numeric value is ℝ(_n_).
```

Comment:
```suggestion
                1. Set _C_ to the code unit whose numeric value is _n_.
```
(`_n_` is already a mathematical value)